### PR TITLE
feat: Hide pets messages

### PR DESCRIPTION
**Pull request by Zffu**
[+] Added the "Hide Pets Messages" Option on the utilities category
[+] The pets messages (example: pet: *test*) will not be displayed if the "Hide Pets messages" option is enabled

### CODE REVIEW

File: src/main/java/com/wynntils/modules/chat/events/ClientEvents.java [hunk 1]
```diff
@@ -97,6 +98,8 @@ public void onChatReceived(ClientChatReceivedEvent e) {
                     message.getStyle().setHoverEvent(e.getMessage().getStyle().getHoverEvent());
                 e.setMessage(message);
             }
+        } else if((UtilitiesConfig.INSTANCE.hidePetsMessages) && (McIf.getUnformattedText(e.getMessage()).matches("(.*): *(.*)*"))){
```
Comment (DevScyu) [inline src/main/java/com/wynntils/modules/chat/events/ClientEvents.java]: This regex is not specific enough and will match more than intended.

Comment (Zffu) [reply src/main/java/com/wynntils/modules/chat/events/ClientEvents.java]: i just realised that a few minutes ago im gonna fix this when im home

**Comment by Zffu** [comment]
fixed

File: src/main/java/com/wynntils/modules/chat/events/ClientEvents.java [hunk 2]
```diff
@@ -97,6 +99,8 @@ public void onChatReceived(ClientChatReceivedEvent e) {
                     message.getStyle().setHoverEvent(e.getMessage().getStyle().getHoverEvent());
                 e.setMessage(message);
             }
+        } else if((UtilitiesConfig.INSTANCE.hidePetsMessages) && (PET_MESSAGE.matcher(McIf.getFormattedText(e.getMessage())).matches())) {
```
Comment (ryanzhoudev) [inline src/main/java/com/wynntils/modules/chat/events/ClientEvents.java]: Please fix the formatting and redundant parentheses.

File: src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java [hunk 3]
```diff
@@ -169,6 +169,9 @@ public class UtilitiesConfig extends SettingsClass {
     @Setting(displayName = "Reverse Ingredient Pouch Order", description = "Should the ingredient pouch sort order be reversed?\n\n§8Click on the ingredient pouch to refresh.", order = 34)
     public boolean sortIngredientPouchReverse = false;
 
+    @Setting(displayName = "Hide Pet Messages", description = "Should Wynntils Hide the Messages from pets", order = 35)
```
Comment (ryanzhoudev) [inline src/main/java/com/wynntils/modules/utilities/configs/UtilitiesConfig.java]:
```suggestion
    @Setting(displayName = "Hide Pet Messages", description = "Should Wynntils hide the messages from pets?", order = 35)
```

**Review by ryanzhoudev — CHANGES_REQUESTED** [review]

File: src/main/java/com/wynntils/modules/chat/events/ClientEvents.java [hunk 4]
```diff
@@ -52,6 +53,7 @@ public class ClientEvents implements Listener {
 
     private static final Pattern GUILD_RESOURCE_WARNING = Pattern.compile("\\[INFO\\] Territory .+ is producing more .+");
     private static final Pattern GUILD_CHAT_MESSAGE = Pattern.compile("\\[(★{0,5})(.+)\\] (.+)");
+    private static final Pattern PET_MESSAGE = Pattern.compile("§2(.*): §a*(.*)*");
```
Comment (ryanzhoudev) [inline src/main/java/com/wynntils/modules/chat/events/ClientEvents.java:56]: Also, please fix your regex. There is no need for the capture groups, and this currently matches more than necessary. You can use https://regexr.com/71b0a, I have created a few tests for you, but you should do your own testing as well.

**Comment by ryanzhoudev** [comment]
@Radi0o why did you resolve the regex comment without solving the problem? This cannot be merged until that is fixed

**Comment by Zffu** [comment]
i changed the regex and make some tests ingame and pets messages are not showing up like intended

File: src/main/java/com/wynntils/modules/chat/events/ClientEvents.java (hunk 4 above)
Comment (Zffu) [reply src/main/java/com/wynntils/modules/chat/events/ClientEvents.java:56]: fixed

**Comment by ryanzhoudev** [comment]
> i changed the regex and make some tests ingame and pets messages are not showing up like intended

What does this mean? I am still not sure why you are marking the issue as resolved and saying it is "fixed" when there have been zero changes

**Pull request closed without merging**